src/Period.php

```diff
@@ -436,6 +438,15 @@ public function getPrecisionMask(): int
         return $this->precisionMask;
     }
 
+    public function getIterator()
```

Comment:
First of all, a great feature and it should be added! I'm wondering is using `\DatePeriod` as the iterator, will keep the `Immutable` constraint this package provides? My guess is it does, but I'm not sure without testing.

Reply:
I assumed as well, but to be sure there's now a test to ensure that the dates in the returned `\DatePeriod` are `\DateTimeImmutables` \o/

Reply:
Great!